src/main/java/org/openrewrite/java/testing/assertj/AdoptAssertJDurationAssertions.java

```diff
@@ -0,0 +1,208 @@
+            if (isEqualToArgRaw == 0 && isRelatedToDuration) {
+                String formatted_template = formatTemplate("assertThat(#{any()}).%s();", m.getSimpleName(), asDescription);
+                templateParameters.set(0, assertThatArg);
+                return applyTemplate(ctx, m, formatted_template, templateParameters.toArray());
+            }
+
+            if (GET_NANO_MATCHER.matches(assertThatArg) || GET_SECONDS_MATCHER.matches(assertThatArg)) {
+                Expression assertThatArgSelect = assertThatArg.getSelect();
+                String methodName = assertThatArg.getSimpleName();
+                String formatted_template = formatTemplate("assertThat(#{any()}).%s(#{any()});", methodName, asDescription);
+                templateParameters.set(0, assertThatArgSelect);
+                templateParameters.add(isEqualToArg);
+
+                return applyTemplate(ctx, m, formatted_template, templateParameters.toArray());
+            }
+
+            return m;
+        }
+
+        private J.MethodInvocation simplifyTimeUnits(J.MethodInvocation m, ExecutionContext ctx) {
+            Expression arg = m.getArguments().get(0);
+            Long argValue = SimplifyDurationCreationUnits.getConstantIntegralValue(arg);
+            if (argValue == null) {
+                return m;
+            }
+
+            List<Object> unitInfo = getUnitInfo(m.getSimpleName(), Math.toIntExact(argValue));
+            String methodName = (String) unitInfo.get(0);
+            int methodArg = (int) unitInfo.get(1);
+            if (!(m.getSimpleName().equals(methodName))) {
+                // update method invocation with new name and arg
+                String template = String.format("#{any()}.%s(%d)", methodName, methodArg);
+                return applyTemplate(ctx, m, template, m.getSelect());
+            }
+
+            return m;
+        }
+
+        private static List<Object> getUnitInfo(String name, int argValue) {
+            int timeLength = 60;
+            if (name.equals("hasMillis")) {
+                timeLength = 1000;
+            } else if (name.equals("hasHours")) {
+                timeLength = 24;
+            }
+
+            if (argValue % timeLength == 0) {
+                String newName = METHOD_MAP.get(name);
+                return getUnitInfo(newName, argValue / timeLength);
+            } else {
+                // returning name, newArg
+                return Arrays.asList(name, argValue);
+            }
+        }
+
+        private J.MethodInvocation applyTemplate(ExecutionContext ctx, J.MethodInvocation m, String template, Object... parameters) {
+            return JavaTemplate.builder(template)
+                    .contextSensitive()
+                    .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "assertj-core-3.24"))
+                    .build()
+                    .apply(getCursor(), m.getCoordinates().replace(), parameters);
```

Comment:
@AlekSimpson What you want to do is to retain the whitespace here. Assuming you know that the result has a non-null `select` you can achieve that by doing something like this:
```java
            J.MethodInvocation invocation = JavaTemplate.builder(template)
                    .contextSensitive()
                    .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "assertj-core-3.24"))
                    .build()
                    .apply(getCursor(), m.getCoordinates().replace(), parameters);
            return invocation.getPadding()
                    .withSelect(
                            invocation.getPadding().getSelect()
                                    .withAfter(m.getPadding().getSelect().getAfter())
                    );
```
But it would probably be best if you guard this with null checks for the selects on both elements.

Reply:
This works! Thanks for pointing it out, I'll be sure to add those null checks in.